### ISSUE TITLE
maple is not listed as a device that can have its bootloader unlocked

### ISSUE THREAD

**Issue by jamuir** [sonyxperiadev/device-sony-maple#11]
"Xperia XZ Premium" isn't in this list:

https://developer.sonymobile.com/unlockbootloader/supported-devices

I've tried generating a bootloader unlock code for "Xperia XZ" (suzu), but it did not work.

Any suggestions on how to unlock the maple bootloader?

**Comment by Umeaboy** [comment]
What Android version are you currently on?
Is it the G8141 or the G8142 variant?
What OS are you using? Windows or Linux?
What distribution? Ubuntu? Fedora, Debian........
What version of ADB are you using?
What's the error you're getting in the terminal window?

Add the output pasted in a document that you drag and drop here.

I used the unlock code made for Xperia XZ to a friends Xperia XZ Premium G8141 using Mageia (an RPM based Linux distribution) and I had no problem. I just had to wait for about 10 minutes or so until I got a new prompt saying that the device is unlocked when it rebooted.

**Comment by alviteri** [comment]
@jamuir I've unlocked my XZ Premium chossing Xperia XZ (keyaki) .... the device selection is only for inform you how to procedee then the system send you the correct unlock code for your device basing on your IMEI CODE

**Comment by jamuir** [comment]
@alviteri @Umeaboy :  thanks for your replies.

I was attempting to issue the fastboot unlock command to the XZ Premium when it was connected to an ubuntu machine (16.04 with a 4.4.0 kernel).  I discovered that the reason the unlock command was failing was because the fastboot connection was unstable (thanks for helping to discount my initial guess that the unlock code I had was not valid for an XZ Premium).

I can demonstrate the fastboot instability by repeatedly issuing the command `fastboot getvar product`.  Most of the time the command fails, but occasionally it will succeed (maybe 2/5 times); e.g. :

```
$ fastboot getvar product
getvar:product FAILED (remote: GetVar Variable Not found)
finished. total time: 0.000s
$ fastboot getvar product
getvar:product FAILED (remote: unknown command)
finished. total time: 0.000s
$ fastboot getvar product
product: G8142
finished. total time: 0.000s
```
I've tried various things to eliminate the instability (changing usb cables, trying different usb ports), but it remains.  I've seen the defect on two different ubuntu machines.

Fortunately, I discovered that when I connect to a mac book, the connection is stable, so I was able to  do an unlock (and also flash new images) from there.

I don't know what is going on with the usb connection.  I have an Xperia X device and there is no problem connecting to it.  Using the most recent version of fastboot does not help.  Connecting with adb is also unstable, so it may be some type of usb negotiation error.
